### ISSUE TITLE
two items: [1] stochastic-weight-averaging [2] learning-rate scheduler to support spec-augment

### ISSUE THREAD

**Issue by levhaikin** [SeanNaren/deepspeech.pytorch#504]
Hi Sean,

There are two items I wanted to bring up with you:
1. have you considered adding stochastic-weight-averaging (swa)? I added it to the code, and got more than two percent absolute improvement. In my experiment, I used a model that I already trained for 54 epochs, and then for the last few epochs I manually added swa with frequency of 500 steps and a constant learning-rate of 0.3 (which is very high compared to what I used for the first 54 epochs)
2. regarding the newly added spec-augment, the (according to the original paper) crucial learning-rate schedule wasn't added. are you considering to add it? and - do you have any results when activating spec-augment that you can share/release?

thanks,
Lev.
